File: Find triplets with zero sum - GFG/find-triplets-with-zero-sum.c
// { Driver Code Starts
//Initial Template for C

#include<stdio.h>

 // } Driver Code Ends
//User function Template for C

int findTriplets(int arr[], int n)
    { 
        //Your code here
        int sum=0;
        for(int i=0;i<n;i++){
            for(int j=(i+1);j<n;j++){
                for(int k=(j+1);k<n;k++){
                    sum+=arr[i]+arr[j]+arr[k];
                    if(sum==0){
                        return 1;
                    }
                    sum=0;
            }
        }
        
    }
    return 0;}

// { Driver Code Starts.
int main()
{
    int t;
	scanf("%d", &t);
	while(t--){
    	int n;
    	scanf("%d", &n);
    	int arr[n];
    	for(int i=0;i<n;i++)
    		scanf("%d", &arr[i]);
        if(findTriplets(arr, n))
            printf("1\n");
        else 
            printf("0\n");
	}
    return 0;
}  // } Driver Code Ends

File: Find triplets with zero sum - GFG/find-triplets-with-zero-sum.java
// { Driver Code Starts
import java.util.*;
class Triplets{
	public static void main(String[] args){
		Scanner sc=new Scanner(System.in);
		int t=sc.nextInt();
		while(t-->0){
			int n=sc.nextInt();
			int[] a=new int[n];
			for(int i=0;i<n;i++){
				a[i]=sc.nextInt();
			}
			Solution g=new Solution();
			if(g.findTriplets(a,n))
				System.out.println("1");
			else
				System.out.println("0");
			
		}
	}
}// } Driver Code Ends


/*Complete the function below*/


class Solution
{
    // arr[]: input array
    // n: size of the array
    //Function to find triplets with zero sum.
	public boolean findTriplets(int arr[] , int n)
    {
        //add code here.
        int sum=0;
        for(int i=0;i<n;i++){
            for(int j=(i+1);j<n;j++){
                for(int k=(j+1);k<n;k++){
                    sum+=arr[i]+arr[j]+arr[k];
                    if(sum==0){
                        return true;
                    }
                    sum=0;
            }
        }
        
    }
    return false;
    }
}